Jint/Engine.cs

```diff
@@ -372,6 +385,17 @@ public Completion ExecuteStatement(Statement statement)
                 throw new TimeoutException();
             }
 
+#if NETCOREAPP2_0
+            if (Options._MemoryLimit > 0)
+            {
+                var memoryUsage = GC.GetAllocatedBytesForCurrentThread() - _initialMemoryUsage;
```

Comment:
Might be a bit more cumbersome, but you could avoid cross-compiling by reflecting on the method. This should make it work in all platforms that support the API.

`Func<int> GetAllocatedBytesForCurrentThread = _methodInfo != null ? Delegate.CreateDelegate() : long.MaxValue;`